### PR TITLE
GT-2199 tool options for openers

### PR DESCRIPTION
**Pull request by rachaelblue**
Hey @levieggertcru, let me know what you think about the inheritance/protocol combination on the observer.  I was wondering if there's a cleaner way to distinguish a tract from CYOA rather than checking the observer type 🤔

### CODE REVIEW

File: godtools/App/Flows/ToolScreenShare/ToolScreenShareFlow.swift [hunk 1]
```diff
@@ -12,7 +12,9 @@ import Combine
 
 class ToolScreenShareFlow: Flow {
 
-    private let toolSettingsObserver: ToolSettingsObserver
+    typealias ToolScreenShareSettingsObserver = ToolSettingsObserver & RemoteShareable
```
Comment (levieggertcru) [inline godtools/App/Flows/ToolScreenShare/ToolScreenShareFlow.swift:15]: Oh whoa I didn't know that would work, basically placing both types into a single type.

**Comment by levieggertcru** [comment]
Hey @rachaelblue regarding your question, was their a specific place you had in question for distinguishing the tract from CYOA and making that check?

So far I think this is a good path forward to make it easy to share the tool settings between different tool types.  I like that a tool option visibility is tied to conformance.  This forces the conformance in order to enable a tool option which aids the programmer in correctly wiring that up.

My only concern with sharing the tool options view is if down the road there were more tool types with more varying tool options or if the tool options view started to look different between tool types.  It could then get difficult to maintain the view.  But, hopefully it won't come to that and the view and options will stay pretty consistent.

**Review by levieggertcru — APPROVED** [review]
Hey @rachaelblue this is looking like a great start.  There are 2 areas I'm wondering if we should consolidate some of the logic.  Doesn't need to be in this PR though and can be a separate PR.

I'm thinking these 2 areas.

[] - Add a createToolSettingsObserver to Base MobileContentPagesViewModel which creates the Parent ToolSettingsObserver and observes languages and training tips.  Subclasses can override to return specific subclass types and do additional observing.

[] - Create a flow protocol extension (ToolSettingsNavigationFlow.swift) for presenting tool settings and dismissing tool settings.  We could then share some of that logic between the TractFlow and ChooseYourOwnAdventureFlow when presenting and dismissing ToolSettings.  Similar to the ToolNavigationFlow protocol extension.  Unfortunately there isn't a built-in way to destroy child flows and we have to do that manually.

File: godtools/App/Features/ToolSettings/Presentation/ToolSettings/ToolSettingsViewModel.swift [hunk 2]
```diff
@@ -46,6 +47,7 @@ class ToolSettingsViewModel: ObservableObject {
         self.viewToolSettingsUseCase = viewToolSettingsUseCase
         self.getShareablesUseCase = getShareablesUseCase
         self.getShareableImageUseCase = getShareableImageUseCase
+        self.hidesShareScreenButton = toolSettingsObserver.isRemoteShareable == false
```
[the remark this answers is not in the document]
Comment (rachaelblue) [reply godtools/App/Features/ToolSettings/Presentation/ToolSettings/ToolSettingsViewModel.swift:50]: This was my main concern, if this check was too obscure and needed to be more obvious.  But I guess at this point it doesn't matter to tool settings if it's CYOA or tract.  Maybe more of an issue if the CYOA/tract views do start looking pretty different, like you said.

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/ToolSettings/Presentation/ToolSettings/ToolSettingsViewModel.swift:50]: Ahh I see, right yeah tool settings doesn't really care at this point.  Just that the observer implements RemoteShareable.

**Comment by rachaelblue** [comment]
Thanks @levieggertcru, made a followup ticket for that work https://jira.cru.org/browse/GT-2426

**Pull request merged**